
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>EC numbers &#8212; als_example_ec_2_2_1_6</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=62ba249389abaaa9ffc34bf36a076bdc1d65ee18" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.b7bb847fb20b106c3d81b95245e65545.min.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=f31d14ad54b65d19161ba51d4ffff3a77ae00456"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="EC numbers" href="als_interpro_IPR012782_subset_summary_cleaned.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="index.html">
      
      
      
      <h1 class="site-logo" id="site-title">als_example_ec_2_2_1_6</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1 current active">
                <a class="reference internal" href="#">
                    EC numbers
                </a>
            </li>
        </ul>
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="als_interpro_IPR012782_subset_summary_cleaned.html">
   EC numbers
  </a>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
            <div class="navbar_extra_footer">
            Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
            </div>
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<div class="menu-dropdown menu-dropdown-launch-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Launch interactive content">
      <i class="fas fa-rocket"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="https://mybinder.org/v2/gh/executablebooks/jupyter-book/master?urlpath=tree/als_example_ec_2_2_1_6_summary_cleaned.ipynb"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Launch on Binder"
>
  

<span class="headerbtn__icon-container">
  
    <img src="_static/images/logo_binder.svg">
  </span>
<span class="headerbtn__text-container">Binder</span>
</a>

      </li>
      
    </ul>
  </div>
</div>

<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>

<div class="menu-dropdown menu-dropdown-download-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Download this page">
      <i class="fas fa-download"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="_sources/als_example_ec_2_2_1_6_summary_cleaned.ipynb"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Download source file"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="headerbtn__text-container">.ipynb</span>
</a>

      </li>
      
      <li>
        
<button onclick="printPdf(this)"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="left"
title="Print to PDF"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="headerbtn__text-container">.pdf</span>
</button>

      </li>
      
    </ul>
  </div>
</div>
<label for="__page-toc"
  class="headerbtn headerbtn-page-toc"
  
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-list"></i>
  </span>

</label>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
    <div class="tocsection onthispage pt-5 pb-3">
        <i class="fas fa-list"></i> Contents
    </div>
    <nav id="bd-toc-nav" aria-label="Page">
        <ul class="visible nav section-nav flex-column">
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#">
   EC numbers
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#taxonomic-distribution">
   Taxonomic distribution
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#annotation-distributions">
   Annotation distributions
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#experimental-data-brenda">
   Experimental data (BRENDA)
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#structural-information">
   Structural information
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#general-data-set-information">
   General data set information
  </a>
 </li>
</ul>

    </nav>
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>EC numbers</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#">
   EC numbers
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#taxonomic-distribution">
   Taxonomic distribution
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#annotation-distributions">
   Annotation distributions
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#experimental-data-brenda">
   Experimental data (BRENDA)
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#structural-information">
   Structural information
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#general-data-set-information">
   General data set information
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>None
</pre></div>
</div>
</div>
</div>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Summary sheet - als_example_ec_2_2_1_6
</pre></div>
</div>
</div>
</div>
<section class="tex2jax_ignore mathjax_ignore" id="ec-numbers">
<h1>EC numbers<a class="headerlink" href="#ec-numbers" title="Permalink to this headline">#</a></h1>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>The EC numbers found in the current data set are {&#39;2.2.1.6&#39;}
</pre></div>
</div>
</div>
</div>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="taxonomic-distribution">
<h1>Taxonomic distribution<a class="headerlink" href="#taxonomic-distribution" title="Permalink to this headline">#</a></h1>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<img alt="_images/als_example_ec_2_2_1_6_summary_cleaned_9_0.png" src="_images/als_example_ec_2_2_1_6_summary_cleaned_9_0.png" />
<img alt="_images/als_example_ec_2_2_1_6_summary_cleaned_9_1.png" src="_images/als_example_ec_2_2_1_6_summary_cleaned_9_1.png" />
</div>
</div>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="annotation-distributions">
<h1>Annotation distributions<a class="headerlink" href="#annotation-distributions" title="Permalink to this headline">#</a></h1>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<img alt="_images/als_example_ec_2_2_1_6_summary_cleaned_11_0.png" src="_images/als_example_ec_2_2_1_6_summary_cleaned_11_0.png" />
<img alt="_images/als_example_ec_2_2_1_6_summary_cleaned_11_1.png" src="_images/als_example_ec_2_2_1_6_summary_cleaned_11_1.png" />
<img alt="_images/als_example_ec_2_2_1_6_summary_cleaned_11_2.png" src="_images/als_example_ec_2_2_1_6_summary_cleaned_11_2.png" />
</div>
</div>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Sequence statistics for all sequences
Number of sequences : 8.0
Smallest sequnce length : 547.0
Longest sequnce length : 687.0
Average sequnce length : 598.0
</pre></div>
</div>
</div>
</div>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="experimental-data-brenda">
<h1>Experimental data (BRENDA)<a class="headerlink" href="#experimental-data-brenda" title="Permalink to this headline">#</a></h1>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>166
</pre></div>
</div>
<img alt="_images/als_example_ec_2_2_1_6_summary_cleaned_14_1.png" src="_images/als_example_ec_2_2_1_6_summary_cleaned_14_1.png" />
</div>
</div>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="structural-information">
<h1>Structural information<a class="headerlink" href="#structural-information" title="Permalink to this headline">#</a></h1>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>There are 34 total structures across 3 unique sequences
Looking at the first few...
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Entry is titled &quot;Crystal structure of Arabidopsis thaliana Acetohydroxyacid synthase In Complex With A Sulfonylurea Herbicide Chlorimuron Ethyl&quot; and was released on 2006/01/10 

This entry was determined using X-ray diffraction 
The authors are McCourt, J.A.Pang, S.S.King-Scott, J.Guddat, L.W.Duggleby, R.G. 

Assembly information -
Non-preferred form with assembly ID 1 is a homotetramer
Preferred form with assembly ID 2 is a homodimer
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Ligand : MAGNESIUM ION at position 699
Ligand : 2-[[[[(4-CHLORO-6-METHOXY-2-PYRIMIDINYL)AMINO]CARBONYL]AMINO]SULFONYL]BENZOIC ACID ETHYL ESTER at position 695
Ligand : 2-[N-CYCLOHEXYLAMINO]ETHANE SULFONIC ACID at position 696
Ligand : FLAVIN-ADENINE DINUCLEOTIDE at position 701
Ligand : ETHYL DIHYDROGEN DIPHOSPHATE at position 702
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>1ybh chain A has helices at residue ranges [&#39;14-24&#39;, &#39;35-37&#39;, &#39;38-48&#39;, &#39;58-74&#39;, &#39;84-89&#39;, &#39;90-101&#39;, &#39;112-116&#39;, &#39;125-131&#39;, &#39;132-134&#39;, &#39;142-144&#39;, &#39;145-159&#39;, &#39;171-176&#39;, &#39;188-195&#39;, &#39;200-214&#39;, &#39;223-226&#39;, &#39;228-240&#39;, &#39;269-279&#39;, &#39;290-295&#39;, &#39;296-300&#39;, &#39;329-344&#39;, &#39;344-349&#39;, &#39;352-366&#39;, &#39;378-391&#39;, &#39;401-410&#39;, &#39;430-442&#39;, &#39;453-459&#39;, &#39;461-469&#39;, &#39;483-494&#39;, &#39;505-509&#39;, &#39;515-523&#39;, &#39;532-546&#39;, &#39;571-575&#39;] and strands at [&#39;177-178&#39;, &#39;52-53&#39;, &#39;77-81&#39;, &#39;29-32&#39;, &#39;104-110&#39;, &#39;164-170&#39;, &#39;136-139&#39;, &#39;12-13&#39;, &#39;243-245&#39;, &#39;281-285&#39;, &#39;260-263&#39;, &#39;217-221&#39;, &#39;305-310&#39;, &#39;324-327&#39;, &#39;419-420&#39;, &#39;447-452&#39;, &#39;395-398&#39;, &#39;473-479&#39;, &#39;550-555&#39;, &#39;527-530&#39;]
</pre></div>
</div>
<div class="output text_html"><div id="3dmolviewer_1654480644674348"  style="position: relative; width: 640px; height: 480px">
        <p id="3dmolwarning_1654480644674348" style="background-color:#ffcccc;color:black">You appear to be running in JupyterLab (or JavaScript failed to load for some other reason).  You need to install the 3dmol extension: <br>
        <tt>jupyter labextension install jupyterlab_3dmol</tt></p>
        </div>
<script>

var loadScriptAsync = function(uri){
  return new Promise((resolve, reject) => {
    var tag = document.createElement('script');
    tag.src = uri;
    tag.async = true;
    tag.onload = () => {
      resolve();
    };
  var firstScriptTag = document.getElementsByTagName('script')[0];
  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
});
};

if(typeof $3Dmolpromise === 'undefined') {
$3Dmolpromise = null;
  $3Dmolpromise = loadScriptAsync('https://3dmol.org/build/3Dmol.js');
}

var viewer_1654480644674348 = null;
var warn = document.getElementById("3dmolwarning_1654480644674348");
if(warn) {
    warn.parentNode.removeChild(warn);
}
$3Dmolpromise.then(function() {
viewer_1654480644674348 = $3Dmol.createViewer($("#3dmolviewer_1654480644674348"),{backgroundColor:"white"});
$3Dmol.download("pdb:1YBH", viewer_1654480644674348, {}, function() {
viewer_1654480644674348.zoomTo();
	viewer_1654480644674348.setStyle({"cartoon": {"color": "spectrum"}});
viewer_1654480644674348.render();
})
});
</script></div><div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;py3Dmol.view at 0x1101de8c0&gt;
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Entry is titled &quot;Crystal structure of Arabidopsis thaliana Acetohydroxyacid synthase In Complex With A Sulfonylurea Herbicide, Metsulfuron methyl&quot; and was released on 2006/01/17 

This entry was determined using X-ray diffraction 
The authors are McCourt, J.A.Pang, S.S.King-Scott, J.Guddat, L.W.Duggleby, R.G. 

Assembly information -
Non-preferred form with assembly ID 1 is a homotetramer
Preferred form with assembly ID 2 is a homodimer
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Ligand : MAGNESIUM ION at position 699
Ligand : METHYL 2-[({[(4-METHOXY-6-METHYL-1,3,5-TRIAZIN-2-YL)AMINO]CARBONYL}AMINO)SULFONYL]BENZOATE at position 695
Ligand : 2-[N-CYCLOHEXYLAMINO]ETHANE SULFONIC ACID at position 696
Ligand : FLAVIN-ADENINE DINUCLEOTIDE at position 701
Ligand : ETHYL DIHYDROGEN DIPHOSPHATE at position 702
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>1yhy chain A has helices at residue ranges [&#39;14-24&#39;, &#39;35-37&#39;, &#39;38-48&#39;, &#39;58-74&#39;, &#39;84-89&#39;, &#39;90-101&#39;, &#39;112-116&#39;, &#39;125-131&#39;, &#39;142-144&#39;, &#39;145-159&#39;, &#39;171-176&#39;, &#39;188-195&#39;, &#39;200-213&#39;, &#39;223-226&#39;, &#39;228-240&#39;, &#39;269-279&#39;, &#39;290-295&#39;, &#39;312-316&#39;, &#39;329-343&#39;, &#39;344-349&#39;, &#39;352-366&#39;, &#39;378-391&#39;, &#39;401-410&#39;, &#39;430-442&#39;, &#39;453-459&#39;, &#39;461-470&#39;, &#39;483-494&#39;, &#39;505-509&#39;, &#39;515-523&#39;, &#39;535-546&#39;, &#39;571-575&#39;] and strands at [&#39;177-178&#39;, &#39;52-53&#39;, &#39;77-81&#39;, &#39;29-33&#39;, &#39;104-110&#39;, &#39;164-170&#39;, &#39;136-139&#39;, &#39;12-13&#39;, &#39;243-245&#39;, &#39;281-285&#39;, &#39;260-263&#39;, &#39;217-221&#39;, &#39;305-310&#39;, &#39;324-327&#39;, &#39;419-420&#39;, &#39;447-452&#39;, &#39;395-398&#39;, &#39;474-479&#39;, &#39;550-555&#39;, &#39;527-530&#39;]
</pre></div>
</div>
<div class="output text_html"><div id="3dmolviewer_1654480648461098"  style="position: relative; width: 640px; height: 480px">
        <p id="3dmolwarning_1654480648461098" style="background-color:#ffcccc;color:black">You appear to be running in JupyterLab (or JavaScript failed to load for some other reason).  You need to install the 3dmol extension: <br>
        <tt>jupyter labextension install jupyterlab_3dmol</tt></p>
        </div>
<script>

var loadScriptAsync = function(uri){
  return new Promise((resolve, reject) => {
    var tag = document.createElement('script');
    tag.src = uri;
    tag.async = true;
    tag.onload = () => {
      resolve();
    };
  var firstScriptTag = document.getElementsByTagName('script')[0];
  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
});
};

if(typeof $3Dmolpromise === 'undefined') {
$3Dmolpromise = null;
  $3Dmolpromise = loadScriptAsync('https://3dmol.org/build/3Dmol.js');
}

var viewer_1654480648461098 = null;
var warn = document.getElementById("3dmolwarning_1654480648461098");
if(warn) {
    warn.parentNode.removeChild(warn);
}
$3Dmolpromise.then(function() {
viewer_1654480648461098 = $3Dmol.createViewer($("#3dmolviewer_1654480648461098"),{backgroundColor:"white"});
$3Dmol.download("pdb:1YHY", viewer_1654480648461098, {}, function() {
viewer_1654480648461098.zoomTo();
	viewer_1654480648461098.setStyle({"cartoon": {"color": "spectrum"}});
viewer_1654480648461098.render();
})
});
</script></div><div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;py3Dmol.view at 0x110359ed0&gt;
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Entry is titled &quot;Crystal structure of Arabidopsis thaliana Acetohydroxyacid synthase In Complex With A Sulfonylurea Herbicide, Chlorsulfuron&quot; and was released on 2006/01/17 

This entry was determined using X-ray diffraction 
The authors are McCourt, J.A.Pang, S.S.King-Scott, J.Guddat, L.W.Duggleby, R.G. 

Assembly information -
Non-preferred form with assembly ID 1 is a homotetramer
Preferred form with assembly ID 2 is a homodimer
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Ligand : MAGNESIUM ION at position 699
Ligand : 1-(2-CHLOROPHENYLSULFONYL)-3-(4-METHOXY-6-METHYL-L,3,5-TRIAZIN-2-YL)UREA at position 695
Ligand : ETHYL DIHYDROGEN DIPHOSPHATE at position 702
Ligand : FLAVIN-ADENINE DINUCLEOTIDE at position 701
Ligand : 2-[N-CYCLOHEXYLAMINO]ETHANE SULFONIC ACID at position 696
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>1yhz chain A has helices at residue ranges [&#39;14-24&#39;, &#39;35-37&#39;, &#39;38-48&#39;, &#39;58-74&#39;, &#39;84-89&#39;, &#39;90-101&#39;, &#39;112-116&#39;, &#39;125-131&#39;, &#39;132-134&#39;, &#39;142-144&#39;, &#39;145-159&#39;, &#39;171-176&#39;, &#39;188-195&#39;, &#39;200-214&#39;, &#39;223-226&#39;, &#39;228-240&#39;, &#39;269-278&#39;, &#39;290-295&#39;, &#39;312-316&#39;, &#39;329-344&#39;, &#39;344-349&#39;, &#39;352-366&#39;, &#39;378-391&#39;, &#39;401-410&#39;, &#39;430-442&#39;, &#39;453-459&#39;, &#39;461-469&#39;, &#39;483-494&#39;, &#39;505-509&#39;, &#39;515-523&#39;, &#39;535-546&#39;, &#39;571-575&#39;] and strands at [&#39;177-178&#39;, &#39;52-53&#39;, &#39;77-81&#39;, &#39;29-33&#39;, &#39;104-110&#39;, &#39;164-170&#39;, &#39;136-139&#39;, &#39;12-13&#39;, &#39;243-245&#39;, &#39;281-285&#39;, &#39;260-263&#39;, &#39;217-221&#39;, &#39;305-310&#39;, &#39;324-327&#39;, &#39;419-420&#39;, &#39;447-452&#39;, &#39;395-398&#39;, &#39;473-479&#39;, &#39;550-555&#39;, &#39;527-530&#39;]
</pre></div>
</div>
<div class="output text_html"><div id="3dmolviewer_1654480652349615"  style="position: relative; width: 640px; height: 480px">
        <p id="3dmolwarning_1654480652349615" style="background-color:#ffcccc;color:black">You appear to be running in JupyterLab (or JavaScript failed to load for some other reason).  You need to install the 3dmol extension: <br>
        <tt>jupyter labextension install jupyterlab_3dmol</tt></p>
        </div>
<script>

var loadScriptAsync = function(uri){
  return new Promise((resolve, reject) => {
    var tag = document.createElement('script');
    tag.src = uri;
    tag.async = true;
    tag.onload = () => {
      resolve();
    };
  var firstScriptTag = document.getElementsByTagName('script')[0];
  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
});
};

if(typeof $3Dmolpromise === 'undefined') {
$3Dmolpromise = null;
  $3Dmolpromise = loadScriptAsync('https://3dmol.org/build/3Dmol.js');
}

var viewer_1654480652349615 = null;
var warn = document.getElementById("3dmolwarning_1654480652349615");
if(warn) {
    warn.parentNode.removeChild(warn);
}
$3Dmolpromise.then(function() {
viewer_1654480652349615 = $3Dmol.createViewer($("#3dmolviewer_1654480652349615"),{backgroundColor:"white"});
$3Dmol.download("pdb:1YHZ", viewer_1654480652349615, {}, function() {
viewer_1654480652349615.zoomTo();
	viewer_1654480652349615.setStyle({"cartoon": {"color": "spectrum"}});
viewer_1654480652349615.render();
})
});
</script></div><div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;py3Dmol.view at 0x110598f70&gt;
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Entry is titled &quot;Crystal structure of Arabidopsis thaliana Acetohydroxyacid synthase In Complex With A Sulfonylurea Herbicide, Sulfometuron methyl&quot; and was released on 2006/01/17 

This entry was determined using X-ray diffraction 
The authors are McCourt, J.A.Pang, S.S.King-Scott, J.Guddat, L.W.Duggleby, R.G. 

Assembly information -
Non-preferred form with assembly ID 1 is a homotetramer
Preferred form with assembly ID 2 is a homodimer
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Ligand : MAGNESIUM ION at position 699
Ligand : METHYL 2-[({[(4,6-DIMETHYLPYRIMIDIN-2-YL)AMINO]CARBONYL}AMINO)SULFONYL]BENZOATE at position 695
Ligand : 2-[N-CYCLOHEXYLAMINO]ETHANE SULFONIC ACID at position 696
Ligand : ETHYL DIHYDROGEN DIPHOSPHATE at position 702
Ligand : FLAVIN-ADENINE DINUCLEOTIDE at position 701
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>1yi0 chain A has helices at residue ranges [&#39;14-24&#39;, &#39;35-37&#39;, &#39;38-48&#39;, &#39;58-74&#39;, &#39;84-89&#39;, &#39;90-100&#39;, &#39;112-116&#39;, &#39;125-131&#39;, &#39;132-134&#39;, &#39;142-144&#39;, &#39;145-159&#39;, &#39;171-176&#39;, &#39;188-195&#39;, &#39;200-214&#39;, &#39;223-226&#39;, &#39;228-240&#39;, &#39;269-278&#39;, &#39;290-295&#39;, &#39;329-344&#39;, &#39;344-349&#39;, &#39;352-366&#39;, &#39;378-391&#39;, &#39;401-410&#39;, &#39;430-442&#39;, &#39;453-459&#39;, &#39;461-469&#39;, &#39;483-494&#39;, &#39;505-509&#39;, &#39;515-523&#39;, &#39;535-546&#39;, &#39;571-575&#39;] and strands at [&#39;177-178&#39;, &#39;52-53&#39;, &#39;77-81&#39;, &#39;29-32&#39;, &#39;104-110&#39;, &#39;164-170&#39;, &#39;136-139&#39;, &#39;12-13&#39;, &#39;243-245&#39;, &#39;281-285&#39;, &#39;260-263&#39;, &#39;217-221&#39;, &#39;305-310&#39;, &#39;324-327&#39;, &#39;419-420&#39;, &#39;447-452&#39;, &#39;395-398&#39;, &#39;473-479&#39;, &#39;550-555&#39;, &#39;527-530&#39;]
</pre></div>
</div>
<div class="output text_html"><div id="3dmolviewer_1654480994998768"  style="position: relative; width: 640px; height: 480px">
        <p id="3dmolwarning_1654480994998768" style="background-color:#ffcccc;color:black">You appear to be running in JupyterLab (or JavaScript failed to load for some other reason).  You need to install the 3dmol extension: <br>
        <tt>jupyter labextension install jupyterlab_3dmol</tt></p>
        </div>
<script>

var loadScriptAsync = function(uri){
  return new Promise((resolve, reject) => {
    var tag = document.createElement('script');
    tag.src = uri;
    tag.async = true;
    tag.onload = () => {
      resolve();
    };
  var firstScriptTag = document.getElementsByTagName('script')[0];
  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
});
};

if(typeof $3Dmolpromise === 'undefined') {
$3Dmolpromise = null;
  $3Dmolpromise = loadScriptAsync('https://3dmol.org/build/3Dmol.js');
}

var viewer_1654480994998768 = null;
var warn = document.getElementById("3dmolwarning_1654480994998768");
if(warn) {
    warn.parentNode.removeChild(warn);
}
$3Dmolpromise.then(function() {
viewer_1654480994998768 = $3Dmol.createViewer($("#3dmolviewer_1654480994998768"),{backgroundColor:"white"});
$3Dmol.download("pdb:1YI0", viewer_1654480994998768, {}, function() {
viewer_1654480994998768.zoomTo();
	viewer_1654480994998768.setStyle({"cartoon": {"color": "spectrum"}});
viewer_1654480994998768.render();
})
});
</script></div><div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;py3Dmol.view at 0x1105b2b30&gt;
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Entry is titled &quot;Crystal structure of Arabidopsis thaliana Acetohydroxyacid synthase In Complex With A Sulfonylurea Herbicide, Tribenuron methyl&quot; and was released on 2006/01/17 

This entry was determined using X-ray diffraction 
The authors are McCourt, J.A.Pang, S.S.King-Scott, J.Guddat, L.W.Duggleby, R.G. 

Assembly information -
Non-preferred form with assembly ID 1 is a homotetramer
Preferred form with assembly ID 2 is a homodimer
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Ligand : MAGNESIUM ION at position 699
Ligand : METHYL 2-[4-METHOXY-6-METHYL-1,3,5-TRAZIN-2-YL(METHYL)CARBAMOYLSULFAMOYL]BENZOATE at position 695
Ligand : 2-[N-CYCLOHEXYLAMINO]ETHANE SULFONIC ACID at position 696
Ligand : ETHYL DIHYDROGEN DIPHOSPHATE at position 700
Ligand : FLAVIN-ADENINE DINUCLEOTIDE at position 701
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>1yi1 chain A has helices at residue ranges [&#39;14-24&#39;, &#39;35-37&#39;, &#39;38-47&#39;, &#39;58-74&#39;, &#39;84-89&#39;, &#39;90-101&#39;, &#39;112-116&#39;, &#39;125-131&#39;, &#39;132-134&#39;, &#39;142-144&#39;, &#39;145-159&#39;, &#39;171-176&#39;, &#39;188-194&#39;, &#39;200-213&#39;, &#39;223-226&#39;, &#39;228-240&#39;, &#39;269-278&#39;, &#39;290-295&#39;, &#39;296-300&#39;, &#39;312-316&#39;, &#39;329-344&#39;, &#39;344-349&#39;, &#39;352-366&#39;, &#39;378-391&#39;, &#39;401-410&#39;, &#39;430-442&#39;, &#39;453-459&#39;, &#39;461-469&#39;, &#39;483-494&#39;, &#39;505-509&#39;, &#39;515-523&#39;, &#39;532-534&#39;, &#39;535-546&#39;, &#39;571-575&#39;] and strands at [&#39;177-178&#39;, &#39;52-53&#39;, &#39;77-81&#39;, &#39;29-32&#39;, &#39;104-110&#39;, &#39;164-170&#39;, &#39;136-139&#39;, &#39;12-13&#39;, &#39;243-245&#39;, &#39;281-285&#39;, &#39;260-263&#39;, &#39;217-221&#39;, &#39;305-310&#39;, &#39;324-327&#39;, &#39;419-420&#39;, &#39;447-452&#39;, &#39;395-398&#39;, &#39;473-479&#39;, &#39;550-555&#39;, &#39;527-530&#39;]
</pre></div>
</div>
<div class="output text_html"><div id="3dmolviewer_1654480999963952"  style="position: relative; width: 640px; height: 480px">
        <p id="3dmolwarning_1654480999963952" style="background-color:#ffcccc;color:black">You appear to be running in JupyterLab (or JavaScript failed to load for some other reason).  You need to install the 3dmol extension: <br>
        <tt>jupyter labextension install jupyterlab_3dmol</tt></p>
        </div>
<script>

var loadScriptAsync = function(uri){
  return new Promise((resolve, reject) => {
    var tag = document.createElement('script');
    tag.src = uri;
    tag.async = true;
    tag.onload = () => {
      resolve();
    };
  var firstScriptTag = document.getElementsByTagName('script')[0];
  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
});
};

if(typeof $3Dmolpromise === 'undefined') {
$3Dmolpromise = null;
  $3Dmolpromise = loadScriptAsync('https://3dmol.org/build/3Dmol.js');
}

var viewer_1654480999963952 = null;
var warn = document.getElementById("3dmolwarning_1654480999963952");
if(warn) {
    warn.parentNode.removeChild(warn);
}
$3Dmolpromise.then(function() {
viewer_1654480999963952 = $3Dmol.createViewer($("#3dmolviewer_1654480999963952"),{backgroundColor:"white"});
$3Dmol.download("pdb:1YI1", viewer_1654480999963952, {}, function() {
viewer_1654480999963952.zoomTo();
	viewer_1654480999963952.setStyle({"cartoon": {"color": "spectrum"}});
viewer_1654480999963952.render();
})
});
</script></div><div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;py3Dmol.view at 0x110299510&gt;
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Entry is titled &quot;Crystal structure of Arabidopsis thaliana Acetohydroxyacid synthase In Complex With An Imidazolinone Herbicide, Imazaquin&quot; and was released on 2006/01/17 

This entry was determined using X-ray diffraction 
The authors are McCourt, J.A.Pang, S.S.King-Scott, J.Guddat, L.W.Duggleby, R.G. 

Assembly information -
Non-preferred form with assembly ID 1 is a homotetramer
Preferred form with assembly ID 2 is a homodimer
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Ligand : MAGNESIUM ION at position 699
Ligand : 2-(4-ISOPROPYL-4-METHYL-5-OXO-4,5-DIHYDRO-1H-IMIDAZOL-2-YL)QUINOLINE-3-CARBOXYLIC ACID at position 694
Ligand : 2-(4-ISOPROPYL-4-METHYL-5-OXO-4,5-DIHYDRO-1H-IMIDAZOL-2-YL)QUINOLINE-3-CARBOXYLIC ACID at position 695
Ligand : 2-[N-CYCLOHEXYLAMINO]ETHANE SULFONIC ACID at position 696
Ligand : FLAVIN-ADENINE DINUCLEOTIDE at position 701
Ligand : THIAMINE DIPHOSPHATE at position 702
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>1z8n chain A has helices at residue ranges [&#39;14-24&#39;, &#39;35-37&#39;, &#39;38-47&#39;, &#39;58-73&#39;, &#39;84-89&#39;, &#39;90-100&#39;, &#39;112-116&#39;, &#39;125-130&#39;, &#39;142-144&#39;, &#39;145-159&#39;, &#39;171-176&#39;, &#39;188-193&#39;, &#39;200-213&#39;, &#39;223-226&#39;, &#39;228-240&#39;, &#39;247-250&#39;, &#39;269-278&#39;, &#39;290-295&#39;, &#39;312-316&#39;, &#39;329-344&#39;, &#39;344-349&#39;, &#39;352-366&#39;, &#39;378-390&#39;, &#39;401-410&#39;, &#39;430-442&#39;, &#39;453-458&#39;, &#39;461-469&#39;, &#39;483-494&#39;, &#39;505-509&#39;, &#39;515-523&#39;, &#39;532-534&#39;, &#39;535-546&#39;, &#39;571-575&#39;] and strands at [&#39;177-178&#39;, &#39;52-53&#39;, &#39;77-81&#39;, &#39;29-33&#39;, &#39;104-110&#39;, &#39;164-170&#39;, &#39;136-139&#39;, &#39;12-13&#39;, &#39;243-245&#39;, &#39;281-285&#39;, &#39;260-263&#39;, &#39;217-221&#39;, &#39;305-310&#39;, &#39;324-327&#39;, &#39;419-420&#39;, &#39;447-452&#39;, &#39;395-398&#39;, &#39;473-479&#39;, &#39;550-555&#39;, &#39;527-530&#39;]
</pre></div>
</div>
<div class="output text_html"><div id="3dmolviewer_16544810038385541"  style="position: relative; width: 640px; height: 480px">
        <p id="3dmolwarning_16544810038385541" style="background-color:#ffcccc;color:black">You appear to be running in JupyterLab (or JavaScript failed to load for some other reason).  You need to install the 3dmol extension: <br>
        <tt>jupyter labextension install jupyterlab_3dmol</tt></p>
        </div>
<script>

var loadScriptAsync = function(uri){
  return new Promise((resolve, reject) => {
    var tag = document.createElement('script');
    tag.src = uri;
    tag.async = true;
    tag.onload = () => {
      resolve();
    };
  var firstScriptTag = document.getElementsByTagName('script')[0];
  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
});
};

if(typeof $3Dmolpromise === 'undefined') {
$3Dmolpromise = null;
  $3Dmolpromise = loadScriptAsync('https://3dmol.org/build/3Dmol.js');
}

var viewer_16544810038385541 = null;
var warn = document.getElementById("3dmolwarning_16544810038385541");
if(warn) {
    warn.parentNode.removeChild(warn);
}
$3Dmolpromise.then(function() {
viewer_16544810038385541 = $3Dmol.createViewer($("#3dmolviewer_16544810038385541"),{backgroundColor:"white"});
$3Dmol.download("pdb:1Z8N", viewer_16544810038385541, {}, function() {
viewer_16544810038385541.zoomTo();
	viewer_16544810038385541.setStyle({"cartoon": {"color": "spectrum"}});
viewer_16544810038385541.render();
})
});
</script></div><div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;py3Dmol.view at 0x1104130a0&gt;
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Entry is titled &quot;Arabidopsis thaliana acetohydroxyacid synthase in complex with monosulfuron&quot; and was released on 2009/03/31 

This entry was determined using X-ray diffraction 
The authors are Guddat, L.W.Duggleby, R.G.Wang, J.-G.Li, Z.-M. 

Assembly information -
Preferred form with assembly ID 1 is a homodimer
Non-preferred form with assembly ID 2 is a homotetramer
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Ligand : N-[(4-methylpyrimidin-2-yl)carbamoyl]-2-nitrobenzenesulfonamide at position 695
Ligand : 2-[N-CYCLOHEXYLAMINO]ETHANE SULFONIC ACID at position 696
Ligand : MAGNESIUM ION at position 699
Ligand : 2-[(2E)-3-[(4-AMINO-2-METHYLPYRIMIDIN-5-YL)METHYL]-2-(1-HYDROXYETHYLIDENE)-4-METHYL-2,3-DIHYDRO-1,3-THIAZOL-5-YL]ETHYL
TRIHYDROGEN DIPHOSPHATE at position 700
Ligand : FLAVIN-ADENINE DINUCLEOTIDE-N5-ISOBUTYL KETONE at position 701
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>3e9y chain A has helices at residue ranges [&#39;13-23&#39;, &#39;34-36&#39;, &#39;37-46&#39;, &#39;57-71&#39;, &#39;83-88&#39;, &#39;89-99&#39;, &#39;111-115&#39;, &#39;124-130&#39;, &#39;131-133&#39;, &#39;141-143&#39;, &#39;144-158&#39;, &#39;170-175&#39;, &#39;187-193&#39;, &#39;199-213&#39;, &#39;222-225&#39;, &#39;227-239&#39;, &#39;246-249&#39;, &#39;268-278&#39;, &#39;289-294&#39;, &#39;295-299&#39;, &#39;328-343&#39;, &#39;343-348&#39;, &#39;351-365&#39;, &#39;377-390&#39;, &#39;400-410&#39;, &#39;429-441&#39;, &#39;452-458&#39;, &#39;461-468&#39;, &#39;482-493&#39;, &#39;514-522&#39;, &#39;534-545&#39;, &#39;570-574&#39;] and strands at [&#39;176-177&#39;, &#39;51-52&#39;, &#39;76-80&#39;, &#39;28-31&#39;, &#39;103-109&#39;, &#39;163-169&#39;, &#39;135-138&#39;, &#39;11-12&#39;, &#39;242-244&#39;, &#39;280-284&#39;, &#39;259-262&#39;, &#39;216-220&#39;, &#39;304-309&#39;, &#39;323-326&#39;, &#39;418-419&#39;, &#39;446-451&#39;, &#39;394-397&#39;, &#39;472-478&#39;, &#39;549-554&#39;, &#39;526-529&#39;]
</pre></div>
</div>
<div class="output text_html"><div id="3dmolviewer_1654481007639307"  style="position: relative; width: 640px; height: 480px">
        <p id="3dmolwarning_1654481007639307" style="background-color:#ffcccc;color:black">You appear to be running in JupyterLab (or JavaScript failed to load for some other reason).  You need to install the 3dmol extension: <br>
        <tt>jupyter labextension install jupyterlab_3dmol</tt></p>
        </div>
<script>

var loadScriptAsync = function(uri){
  return new Promise((resolve, reject) => {
    var tag = document.createElement('script');
    tag.src = uri;
    tag.async = true;
    tag.onload = () => {
      resolve();
    };
  var firstScriptTag = document.getElementsByTagName('script')[0];
  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
});
};

if(typeof $3Dmolpromise === 'undefined') {
$3Dmolpromise = null;
  $3Dmolpromise = loadScriptAsync('https://3dmol.org/build/3Dmol.js');
}

var viewer_1654481007639307 = null;
var warn = document.getElementById("3dmolwarning_1654481007639307");
if(warn) {
    warn.parentNode.removeChild(warn);
}
$3Dmolpromise.then(function() {
viewer_1654481007639307 = $3Dmol.createViewer($("#3dmolviewer_1654481007639307"),{backgroundColor:"white"});
$3Dmol.download("pdb:3E9Y", viewer_1654481007639307, {}, function() {
viewer_1654481007639307.zoomTo();
	viewer_1654481007639307.setStyle({"cartoon": {"color": "spectrum"}});
viewer_1654481007639307.render();
})
});
</script></div><div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;py3Dmol.view at 0x110359ed0&gt;
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Entry is titled &quot;Arabidopsis thaliana acetohydroxyacid synthase in complex with monosulfuron-ester&quot; and was released on 2009/03/31 

This entry was determined using X-ray diffraction 
The authors are Guddat, L.W.Wang, J.-G.Li, Z.-M. 

Assembly information -
Preferred form with assembly ID 1 is a homodimer
Non-preferred form with assembly ID 2 is a homotetramer
</pre></div>
</div>
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">KeyboardInterrupt</span><span class="g g-Whitespace">                         </span>Traceback (most recent call last)
<span class="nn">Input In [12],</span> in <span class="ni">&lt;cell line: 14&gt;</span><span class="nt">()</span>
<span class="g g-Whitespace">     </span><span class="mi">19</span> <span class="nb">print</span><span class="p">(</span><span class="n">make_summary</span><span class="p">(</span><span class="n">get_entry_from_api</span><span class="p">(</span><span class="n">pdb_id</span><span class="p">,</span> <span class="n">summary_url</span><span class="p">)))</span>
<span class="g g-Whitespace">     </span><span class="mi">21</span> <span class="n">display</span><span class="p">(</span><span class="n">Markdown</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;# </span><span class="si">{</span><span class="n">pdb_id</span><span class="si">}</span><span class="s1"> Ligand informtion&#39;</span><span class="p">))</span>    
<span class="ne">---&gt; </span><span class="mi">22</span> <span class="nb">print</span><span class="p">(</span><span class="n">get_ligand_information</span><span class="p">(</span><span class="n">get_entry_from_api</span><span class="p">(</span><span class="n">pdb_id</span><span class="p">,</span> <span class="n">ligand_url</span><span class="p">)))</span>
<span class="g g-Whitespace">     </span><span class="mi">25</span> <span class="n">display</span><span class="p">(</span><span class="n">Markdown</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;# </span><span class="si">{</span><span class="n">pdb_id</span><span class="si">}</span><span class="s1"> Secondary structure ranges&#39;</span><span class="p">))</span>
<span class="g g-Whitespace">     </span><span class="mi">26</span> <span class="n">get_secondary_structure_ranges</span><span class="p">(</span><span class="n">pdb_id</span><span class="p">)</span>

<span class="nn">Input In [3],</span> in <span class="ni">get_entry_from_api</span><span class="nt">(pdb_id, api_url)</span>
<span class="g g-Whitespace">    </span><span class="mi">203</span>     <span class="k">return</span> <span class="kc">None</span>
<span class="g g-Whitespace">    </span><span class="mi">205</span> <span class="c1"># Make a GET call to the API URL</span>
<span class="ne">--&gt; </span><span class="mi">206</span> <span class="n">get_request</span> <span class="o">=</span> <span class="n">requests</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="n">url</span><span class="o">=</span><span class="n">api_url</span><span class="o">+</span><span class="n">pdb_id</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">208</span> <span class="k">if</span> <span class="n">get_request</span><span class="o">.</span><span class="n">status_code</span> <span class="o">==</span> <span class="mi">200</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">209</span>     <span class="c1"># If there is data returned (with HTML status code 200)</span>
<span class="g g-Whitespace">    </span><span class="mi">210</span>     <span class="c1"># then return the data in JSON format</span>
<span class="g g-Whitespace">    </span><span class="mi">211</span>     <span class="k">return</span> <span class="n">get_request</span><span class="o">.</span><span class="n">json</span><span class="p">()</span>

<span class="nn">File ~/opt/miniconda3/envs/cleanrepo/lib/python3.10/site-packages/requests/api.py:75,</span> in <span class="ni">get</span><span class="nt">(url, params, **kwargs)</span>
<span class="g g-Whitespace">     </span><span class="mi">64</span> <span class="k">def</span> <span class="nf">get</span><span class="p">(</span><span class="n">url</span><span class="p">,</span> <span class="n">params</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="o">**</span><span class="n">kwargs</span><span class="p">):</span>
<span class="g g-Whitespace">     </span><span class="mi">65</span>     <span class="sa">r</span><span class="sd">&quot;&quot;&quot;Sends a GET request.</span>
<span class="g g-Whitespace">     </span><span class="mi">66</span><span class="sd"> </span>
<span class="g g-Whitespace">     </span><span class="mi">67</span><span class="sd">     :param url: URL for the new :class:`Request` object.</span>
<span class="sd">   (...)</span>
<span class="g g-Whitespace">     </span><span class="mi">72</span><span class="sd">     :rtype: requests.Response</span>
<span class="g g-Whitespace">     </span><span class="mi">73</span><span class="sd">     &quot;&quot;&quot;</span>
<span class="ne">---&gt; </span><span class="mi">75</span>     <span class="k">return</span> <span class="n">request</span><span class="p">(</span><span class="s1">&#39;get&#39;</span><span class="p">,</span> <span class="n">url</span><span class="p">,</span> <span class="n">params</span><span class="o">=</span><span class="n">params</span><span class="p">,</span> <span class="o">**</span><span class="n">kwargs</span><span class="p">)</span>

<span class="nn">File ~/opt/miniconda3/envs/cleanrepo/lib/python3.10/site-packages/requests/api.py:61,</span> in <span class="ni">request</span><span class="nt">(method, url, **kwargs)</span>
<span class="g g-Whitespace">     </span><span class="mi">57</span> <span class="c1"># By using the &#39;with&#39; statement we are sure the session is closed, thus we</span>
<span class="g g-Whitespace">     </span><span class="mi">58</span> <span class="c1"># avoid leaving sockets open which can trigger a ResourceWarning in some</span>
<span class="g g-Whitespace">     </span><span class="mi">59</span> <span class="c1"># cases, and look like a memory leak in others.</span>
<span class="g g-Whitespace">     </span><span class="mi">60</span> <span class="k">with</span> <span class="n">sessions</span><span class="o">.</span><span class="n">Session</span><span class="p">()</span> <span class="k">as</span> <span class="n">session</span><span class="p">:</span>
<span class="ne">---&gt; </span><span class="mi">61</span>     <span class="k">return</span> <span class="n">session</span><span class="o">.</span><span class="n">request</span><span class="p">(</span><span class="n">method</span><span class="o">=</span><span class="n">method</span><span class="p">,</span> <span class="n">url</span><span class="o">=</span><span class="n">url</span><span class="p">,</span> <span class="o">**</span><span class="n">kwargs</span><span class="p">)</span>

<span class="nn">File ~/opt/miniconda3/envs/cleanrepo/lib/python3.10/site-packages/requests/sessions.py:529,</span> in <span class="ni">Session.request</span><span class="nt">(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)</span>
<span class="g g-Whitespace">    </span><span class="mi">524</span> <span class="n">send_kwargs</span> <span class="o">=</span> <span class="p">{</span>
<span class="g g-Whitespace">    </span><span class="mi">525</span>     <span class="s1">&#39;timeout&#39;</span><span class="p">:</span> <span class="n">timeout</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">526</span>     <span class="s1">&#39;allow_redirects&#39;</span><span class="p">:</span> <span class="n">allow_redirects</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">527</span> <span class="p">}</span>
<span class="g g-Whitespace">    </span><span class="mi">528</span> <span class="n">send_kwargs</span><span class="o">.</span><span class="n">update</span><span class="p">(</span><span class="n">settings</span><span class="p">)</span>
<span class="ne">--&gt; </span><span class="mi">529</span> <span class="n">resp</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">send</span><span class="p">(</span><span class="n">prep</span><span class="p">,</span> <span class="o">**</span><span class="n">send_kwargs</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">531</span> <span class="k">return</span> <span class="n">resp</span>

<span class="nn">File ~/opt/miniconda3/envs/cleanrepo/lib/python3.10/site-packages/requests/sessions.py:645,</span> in <span class="ni">Session.send</span><span class="nt">(self, request, **kwargs)</span>
<span class="g g-Whitespace">    </span><span class="mi">642</span> <span class="n">start</span> <span class="o">=</span> <span class="n">preferred_clock</span><span class="p">()</span>
<span class="g g-Whitespace">    </span><span class="mi">644</span> <span class="c1"># Send the request</span>
<span class="ne">--&gt; </span><span class="mi">645</span> <span class="n">r</span> <span class="o">=</span> <span class="n">adapter</span><span class="o">.</span><span class="n">send</span><span class="p">(</span><span class="n">request</span><span class="p">,</span> <span class="o">**</span><span class="n">kwargs</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">647</span> <span class="c1"># Total elapsed time of the request (approximately)</span>
<span class="g g-Whitespace">    </span><span class="mi">648</span> <span class="n">elapsed</span> <span class="o">=</span> <span class="n">preferred_clock</span><span class="p">()</span> <span class="o">-</span> <span class="n">start</span>

<span class="nn">File ~/opt/miniconda3/envs/cleanrepo/lib/python3.10/site-packages/requests/adapters.py:440,</span> in <span class="ni">HTTPAdapter.send</span><span class="nt">(self, request, stream, timeout, verify, cert, proxies)</span>
<span class="g g-Whitespace">    </span><span class="mi">438</span> <span class="k">try</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">439</span>     <span class="k">if</span> <span class="ow">not</span> <span class="n">chunked</span><span class="p">:</span>
<span class="ne">--&gt; </span><span class="mi">440</span>         <span class="n">resp</span> <span class="o">=</span> <span class="n">conn</span><span class="o">.</span><span class="n">urlopen</span><span class="p">(</span>
<span class="g g-Whitespace">    </span><span class="mi">441</span>             <span class="n">method</span><span class="o">=</span><span class="n">request</span><span class="o">.</span><span class="n">method</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">442</span>             <span class="n">url</span><span class="o">=</span><span class="n">url</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">443</span>             <span class="n">body</span><span class="o">=</span><span class="n">request</span><span class="o">.</span><span class="n">body</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">444</span>             <span class="n">headers</span><span class="o">=</span><span class="n">request</span><span class="o">.</span><span class="n">headers</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">445</span>             <span class="n">redirect</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">446</span>             <span class="n">assert_same_host</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">447</span>             <span class="n">preload_content</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">448</span>             <span class="n">decode_content</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">449</span>             <span class="n">retries</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">max_retries</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">450</span>             <span class="n">timeout</span><span class="o">=</span><span class="n">timeout</span>
<span class="g g-Whitespace">    </span><span class="mi">451</span>         <span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">453</span>     <span class="c1"># Send the request.</span>
<span class="g g-Whitespace">    </span><span class="mi">454</span>     <span class="k">else</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">455</span>         <span class="k">if</span> <span class="nb">hasattr</span><span class="p">(</span><span class="n">conn</span><span class="p">,</span> <span class="s1">&#39;proxy_pool&#39;</span><span class="p">):</span>

<span class="nn">File ~/opt/miniconda3/envs/cleanrepo/lib/python3.10/site-packages/urllib3/connectionpool.py:703,</span> in <span class="ni">HTTPConnectionPool.urlopen</span><span class="nt">(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)</span>
<span class="g g-Whitespace">    </span><span class="mi">700</span>     <span class="bp">self</span><span class="o">.</span><span class="n">_prepare_proxy</span><span class="p">(</span><span class="n">conn</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">702</span> <span class="c1"># Make the request on the httplib connection object.</span>
<span class="ne">--&gt; </span><span class="mi">703</span> <span class="n">httplib_response</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_make_request</span><span class="p">(</span>
<span class="g g-Whitespace">    </span><span class="mi">704</span>     <span class="n">conn</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">705</span>     <span class="n">method</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">706</span>     <span class="n">url</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">707</span>     <span class="n">timeout</span><span class="o">=</span><span class="n">timeout_obj</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">708</span>     <span class="n">body</span><span class="o">=</span><span class="n">body</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">709</span>     <span class="n">headers</span><span class="o">=</span><span class="n">headers</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">710</span>     <span class="n">chunked</span><span class="o">=</span><span class="n">chunked</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">711</span> <span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">713</span> <span class="c1"># If we&#39;re going to release the connection in ``finally:``, then</span>
<span class="g g-Whitespace">    </span><span class="mi">714</span> <span class="c1"># the response doesn&#39;t need to know about the connection. Otherwise</span>
<span class="g g-Whitespace">    </span><span class="mi">715</span> <span class="c1"># it will also try to release it and we&#39;ll have a double-release</span>
<span class="g g-Whitespace">    </span><span class="mi">716</span> <span class="c1"># mess.</span>
<span class="g g-Whitespace">    </span><span class="mi">717</span> <span class="n">response_conn</span> <span class="o">=</span> <span class="n">conn</span> <span class="k">if</span> <span class="ow">not</span> <span class="n">release_conn</span> <span class="k">else</span> <span class="kc">None</span>

<span class="nn">File ~/opt/miniconda3/envs/cleanrepo/lib/python3.10/site-packages/urllib3/connectionpool.py:386,</span> in <span class="ni">HTTPConnectionPool._make_request</span><span class="nt">(self, conn, method, url, timeout, chunked, **httplib_request_kw)</span>
<span class="g g-Whitespace">    </span><span class="mi">384</span> <span class="c1"># Trigger any extra validation we need to do.</span>
<span class="g g-Whitespace">    </span><span class="mi">385</span> <span class="k">try</span><span class="p">:</span>
<span class="ne">--&gt; </span><span class="mi">386</span>     <span class="bp">self</span><span class="o">.</span><span class="n">_validate_conn</span><span class="p">(</span><span class="n">conn</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">387</span> <span class="k">except</span> <span class="p">(</span><span class="n">SocketTimeout</span><span class="p">,</span> <span class="n">BaseSSLError</span><span class="p">)</span> <span class="k">as</span> <span class="n">e</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">388</span>     <span class="c1"># Py2 raises this as a BaseSSLError, Py3 raises it as socket timeout.</span>
<span class="g g-Whitespace">    </span><span class="mi">389</span>     <span class="bp">self</span><span class="o">.</span><span class="n">_raise_timeout</span><span class="p">(</span><span class="n">err</span><span class="o">=</span><span class="n">e</span><span class="p">,</span> <span class="n">url</span><span class="o">=</span><span class="n">url</span><span class="p">,</span> <span class="n">timeout_value</span><span class="o">=</span><span class="n">conn</span><span class="o">.</span><span class="n">timeout</span><span class="p">)</span>

<span class="nn">File ~/opt/miniconda3/envs/cleanrepo/lib/python3.10/site-packages/urllib3/connectionpool.py:1040,</span> in <span class="ni">HTTPSConnectionPool._validate_conn</span><span class="nt">(self, conn)</span>
<span class="g g-Whitespace">   </span><span class="mi">1038</span> <span class="c1"># Force connect early to allow us to validate the connection.</span>
<span class="g g-Whitespace">   </span><span class="mi">1039</span> <span class="k">if</span> <span class="ow">not</span> <span class="nb">getattr</span><span class="p">(</span><span class="n">conn</span><span class="p">,</span> <span class="s2">&quot;sock&quot;</span><span class="p">,</span> <span class="kc">None</span><span class="p">):</span>  <span class="c1"># AppEngine might not have  `.sock`</span>
<span class="ne">-&gt; </span><span class="mi">1040</span>     <span class="n">conn</span><span class="o">.</span><span class="n">connect</span><span class="p">()</span>
<span class="g g-Whitespace">   </span><span class="mi">1042</span> <span class="k">if</span> <span class="ow">not</span> <span class="n">conn</span><span class="o">.</span><span class="n">is_verified</span><span class="p">:</span>
<span class="g g-Whitespace">   </span><span class="mi">1043</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span>
<span class="g g-Whitespace">   </span><span class="mi">1044</span>         <span class="p">(</span>
<span class="g g-Whitespace">   </span><span class="mi">1045</span>             <span class="s2">&quot;Unverified HTTPS request is being made to host &#39;</span><span class="si">%s</span><span class="s2">&#39;. &quot;</span>
   <span class="p">(</span><span class="o">...</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1050</span>         <span class="n">InsecureRequestWarning</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1051</span>     <span class="p">)</span>

<span class="nn">File ~/opt/miniconda3/envs/cleanrepo/lib/python3.10/site-packages/urllib3/connection.py:414,</span> in <span class="ni">HTTPSConnection.connect</span><span class="nt">(self)</span>
<span class="g g-Whitespace">    </span><span class="mi">405</span> <span class="k">if</span> <span class="p">(</span>
<span class="g g-Whitespace">    </span><span class="mi">406</span>     <span class="ow">not</span> <span class="bp">self</span><span class="o">.</span><span class="n">ca_certs</span>
<span class="g g-Whitespace">    </span><span class="mi">407</span>     <span class="ow">and</span> <span class="ow">not</span> <span class="bp">self</span><span class="o">.</span><span class="n">ca_cert_dir</span>
   <span class="p">(</span><span class="o">...</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">410</span>     <span class="ow">and</span> <span class="nb">hasattr</span><span class="p">(</span><span class="n">context</span><span class="p">,</span> <span class="s2">&quot;load_default_certs&quot;</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">411</span> <span class="p">):</span>
<span class="g g-Whitespace">    </span><span class="mi">412</span>     <span class="n">context</span><span class="o">.</span><span class="n">load_default_certs</span><span class="p">()</span>
<span class="ne">--&gt; </span><span class="mi">414</span> <span class="bp">self</span><span class="o">.</span><span class="n">sock</span> <span class="o">=</span> <span class="n">ssl_wrap_socket</span><span class="p">(</span>
<span class="g g-Whitespace">    </span><span class="mi">415</span>     <span class="n">sock</span><span class="o">=</span><span class="n">conn</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">416</span>     <span class="n">keyfile</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">key_file</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">417</span>     <span class="n">certfile</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">cert_file</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">418</span>     <span class="n">key_password</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">key_password</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">419</span>     <span class="n">ca_certs</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">ca_certs</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">420</span>     <span class="n">ca_cert_dir</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">ca_cert_dir</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">421</span>     <span class="n">ca_cert_data</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">ca_cert_data</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">422</span>     <span class="n">server_hostname</span><span class="o">=</span><span class="n">server_hostname</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">423</span>     <span class="n">ssl_context</span><span class="o">=</span><span class="n">context</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">424</span>     <span class="n">tls_in_tls</span><span class="o">=</span><span class="n">tls_in_tls</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">425</span> <span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">427</span> <span class="c1"># If we&#39;re using all defaults and the connection</span>
<span class="g g-Whitespace">    </span><span class="mi">428</span> <span class="c1"># is TLSv1 or TLSv1.1 we throw a DeprecationWarning</span>
<span class="g g-Whitespace">    </span><span class="mi">429</span> <span class="c1"># for the host.</span>
<span class="g g-Whitespace">    </span><span class="mi">430</span> <span class="k">if</span> <span class="p">(</span>
<span class="g g-Whitespace">    </span><span class="mi">431</span>     <span class="n">default_ssl_context</span>
<span class="g g-Whitespace">    </span><span class="mi">432</span>     <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">ssl_version</span> <span class="ow">is</span> <span class="kc">None</span>
<span class="g g-Whitespace">    </span><span class="mi">433</span>     <span class="ow">and</span> <span class="nb">hasattr</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">sock</span><span class="p">,</span> <span class="s2">&quot;version&quot;</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">434</span>     <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">sock</span><span class="o">.</span><span class="n">version</span><span class="p">()</span> <span class="ow">in</span> <span class="p">{</span><span class="s2">&quot;TLSv1&quot;</span><span class="p">,</span> <span class="s2">&quot;TLSv1.1&quot;</span><span class="p">}</span>
<span class="g g-Whitespace">    </span><span class="mi">435</span> <span class="p">):</span>

<span class="nn">File ~/opt/miniconda3/envs/cleanrepo/lib/python3.10/site-packages/urllib3/util/ssl_.py:449,</span> in <span class="ni">ssl_wrap_socket</span><span class="nt">(sock, keyfile, certfile, cert_reqs, ca_certs, server_hostname, ssl_version, ciphers, ssl_context, ca_cert_dir, key_password, ca_cert_data, tls_in_tls)</span>
<span class="g g-Whitespace">    </span><span class="mi">437</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span>
<span class="g g-Whitespace">    </span><span class="mi">438</span>         <span class="s2">&quot;An HTTPS request has been made, but the SNI (Server Name &quot;</span>
<span class="g g-Whitespace">    </span><span class="mi">439</span>         <span class="s2">&quot;Indication) extension to TLS is not available on this platform. &quot;</span>
   <span class="p">(</span><span class="o">...</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">445</span>         <span class="n">SNIMissingWarning</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">446</span>     <span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">448</span> <span class="k">if</span> <span class="n">send_sni</span><span class="p">:</span>
<span class="ne">--&gt; </span><span class="mi">449</span>     <span class="n">ssl_sock</span> <span class="o">=</span> <span class="n">_ssl_wrap_socket_impl</span><span class="p">(</span>
<span class="g g-Whitespace">    </span><span class="mi">450</span>         <span class="n">sock</span><span class="p">,</span> <span class="n">context</span><span class="p">,</span> <span class="n">tls_in_tls</span><span class="p">,</span> <span class="n">server_hostname</span><span class="o">=</span><span class="n">server_hostname</span>
<span class="g g-Whitespace">    </span><span class="mi">451</span>     <span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">452</span> <span class="k">else</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">453</span>     <span class="n">ssl_sock</span> <span class="o">=</span> <span class="n">_ssl_wrap_socket_impl</span><span class="p">(</span><span class="n">sock</span><span class="p">,</span> <span class="n">context</span><span class="p">,</span> <span class="n">tls_in_tls</span><span class="p">)</span>

<span class="nn">File ~/opt/miniconda3/envs/cleanrepo/lib/python3.10/site-packages/urllib3/util/ssl_.py:493,</span> in <span class="ni">_ssl_wrap_socket_impl</span><span class="nt">(sock, ssl_context, tls_in_tls, server_hostname)</span>
<span class="g g-Whitespace">    </span><span class="mi">490</span>     <span class="k">return</span> <span class="n">SSLTransport</span><span class="p">(</span><span class="n">sock</span><span class="p">,</span> <span class="n">ssl_context</span><span class="p">,</span> <span class="n">server_hostname</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">492</span> <span class="k">if</span> <span class="n">server_hostname</span><span class="p">:</span>
<span class="ne">--&gt; </span><span class="mi">493</span>     <span class="k">return</span> <span class="n">ssl_context</span><span class="o">.</span><span class="n">wrap_socket</span><span class="p">(</span><span class="n">sock</span><span class="p">,</span> <span class="n">server_hostname</span><span class="o">=</span><span class="n">server_hostname</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">494</span> <span class="k">else</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">495</span>     <span class="k">return</span> <span class="n">ssl_context</span><span class="o">.</span><span class="n">wrap_socket</span><span class="p">(</span><span class="n">sock</span><span class="p">)</span>

<span class="nn">File ~/opt/miniconda3/envs/cleanrepo/lib/python3.10/ssl.py:512,</span> in <span class="ni">SSLContext.wrap_socket</span><span class="nt">(self, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, session)</span>
<span class="g g-Whitespace">    </span><span class="mi">506</span> <span class="k">def</span> <span class="nf">wrap_socket</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">sock</span><span class="p">,</span> <span class="n">server_side</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">507</span>                 <span class="n">do_handshake_on_connect</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">508</span>                 <span class="n">suppress_ragged_eofs</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">509</span>                 <span class="n">server_hostname</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">session</span><span class="o">=</span><span class="kc">None</span><span class="p">):</span>
<span class="g g-Whitespace">    </span><span class="mi">510</span>     <span class="c1"># SSLSocket class handles server_hostname encoding before it calls</span>
<span class="g g-Whitespace">    </span><span class="mi">511</span>     <span class="c1"># ctx._wrap_socket()</span>
<span class="ne">--&gt; </span><span class="mi">512</span>     <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">sslsocket_class</span><span class="o">.</span><span class="n">_create</span><span class="p">(</span>
<span class="g g-Whitespace">    </span><span class="mi">513</span>         <span class="n">sock</span><span class="o">=</span><span class="n">sock</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">514</span>         <span class="n">server_side</span><span class="o">=</span><span class="n">server_side</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">515</span>         <span class="n">do_handshake_on_connect</span><span class="o">=</span><span class="n">do_handshake_on_connect</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">516</span>         <span class="n">suppress_ragged_eofs</span><span class="o">=</span><span class="n">suppress_ragged_eofs</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">517</span>         <span class="n">server_hostname</span><span class="o">=</span><span class="n">server_hostname</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">518</span>         <span class="n">context</span><span class="o">=</span><span class="bp">self</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">519</span>         <span class="n">session</span><span class="o">=</span><span class="n">session</span>
<span class="g g-Whitespace">    </span><span class="mi">520</span>     <span class="p">)</span>

<span class="nn">File ~/opt/miniconda3/envs/cleanrepo/lib/python3.10/ssl.py:1070,</span> in <span class="ni">SSLSocket._create</span><span class="nt">(cls, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, context, session)</span>
<span class="g g-Whitespace">   </span><span class="mi">1067</span>         <span class="k">if</span> <span class="n">timeout</span> <span class="o">==</span> <span class="mf">0.0</span><span class="p">:</span>
<span class="g g-Whitespace">   </span><span class="mi">1068</span>             <span class="c1"># non-blocking</span>
<span class="g g-Whitespace">   </span><span class="mi">1069</span>             <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s2">&quot;do_handshake_on_connect should not be specified for non-blocking sockets&quot;</span><span class="p">)</span>
<span class="ne">-&gt; </span><span class="mi">1070</span>         <span class="bp">self</span><span class="o">.</span><span class="n">do_handshake</span><span class="p">()</span>
<span class="g g-Whitespace">   </span><span class="mi">1071</span> <span class="k">except</span> <span class="p">(</span><span class="ne">OSError</span><span class="p">,</span> <span class="ne">ValueError</span><span class="p">):</span>
<span class="g g-Whitespace">   </span><span class="mi">1072</span>     <span class="bp">self</span><span class="o">.</span><span class="n">close</span><span class="p">()</span>

<span class="nn">File ~/opt/miniconda3/envs/cleanrepo/lib/python3.10/ssl.py:1341,</span> in <span class="ni">SSLSocket.do_handshake</span><span class="nt">(self, block)</span>
<span class="g g-Whitespace">   </span><span class="mi">1339</span>     <span class="k">if</span> <span class="n">timeout</span> <span class="o">==</span> <span class="mf">0.0</span> <span class="ow">and</span> <span class="n">block</span><span class="p">:</span>
<span class="g g-Whitespace">   </span><span class="mi">1340</span>         <span class="bp">self</span><span class="o">.</span><span class="n">settimeout</span><span class="p">(</span><span class="kc">None</span><span class="p">)</span>
<span class="ne">-&gt; </span><span class="mi">1341</span>     <span class="bp">self</span><span class="o">.</span><span class="n">_sslobj</span><span class="o">.</span><span class="n">do_handshake</span><span class="p">()</span>
<span class="g g-Whitespace">   </span><span class="mi">1342</span> <span class="k">finally</span><span class="p">:</span>
<span class="g g-Whitespace">   </span><span class="mi">1343</span>     <span class="bp">self</span><span class="o">.</span><span class="n">settimeout</span><span class="p">(</span><span class="n">timeout</span><span class="p">)</span>

<span class="ne">KeyboardInterrupt</span>: 
</pre></div>
</div>
</div>
</div>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="general-data-set-information">
<h1>General data set information<a class="headerlink" href="#general-data-set-information" title="Permalink to this headline">#</a></h1>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<p>These are the columns that are available and have at least one entry</p>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Entry
Truncated_Info
Extracted_ID
Sequence
Original_FASTA
Entry_r
Entry.1
Entry_name
Gene_names
Gene_names__primary_
Gene_names__synonym_
Gene_names__ordered_locus_
Gene_names__ORF_
Organism
Organism_ID
Protein_names
Sequence_r
Proteomes
Taxonomic_lineage_ALL
Erroneous_initiation
Frameshift
Sequence_caution
Length
Mass
Sequence_conflict
Version_sequence
EC_number
Catalytic_activity
ChEBI
ChEBI_Catalytic_activity
ChEBI_Cofactor
ChEBI_IDs
Cofactor
Function_[CC]
Kinetics
Pathway
Temperature_dependence
pH_dependence
Binding_site
Metal_binding
Nucleotide_binding
Annotation
Features
Miscellaneous_[CC]
Keywords
Protein_existence
Status
Subunit_structure_[CC]
Induction
Gene_ontology_GO
Gene_ontology_biological_process
Gene_ontology_molecular_function
Gene_ontology_cellular_component
Gene_ontology_IDs
Biotechnological_use
Disruption_phenotype
Subcellular_location_[CC]
Chain
Initiator_methionine
Modified_residue
Transit_peptide
3D
Beta_strand
Helix
Turn
Mapped_PubMed_ID
PubMed_ID
Date_of_creation
Date_of_last_modification
Date_of_last_sequence_modification
Version_entry
Sequence_similarities
Protein_families
Coiled_coil
Compositional_bias
Region
Taxonomic_lineage_all
Taxonomic_lineage_SUPERKINGDOM
Taxonomic_lineage_KINGDOM
Taxonomic_lineage_SUBKINGDOM
Taxonomic_lineage_PHYLUM
Taxonomic_lineage_SUBPHYLUM
Taxonomic_lineage_CLASS
Taxonomic_lineage_ORDER
Taxonomic_lineage_FAMILY
Taxonomic_lineage_TRIBE
Taxonomic_lineage_GENUS
Taxonomic_lineage_SPECIES_GROUP
Taxonomic_lineage_SPECIES
Taxonomic_lineage_SUBSPECIES
Cross_reference_Araport
Cross_reference_BindingDB
Cross_reference_BioCyc
Cross_reference_BioGRID
Cross_reference_BRENDA
Cross_reference_CarbonylDB
Cross_reference_CDD
Cross_reference_ChEMBL
Cross_reference_ComplexPortal
Cross_reference_DIP
Cross_reference_DisProt
Cross_reference_DNASU
Cross_reference_DrugBank
Cross_reference_EchoBASE
Cross_reference_eggNOG
Cross_reference_EMBL
EnsemblBacteria_transcript
EnsemblFungi_transcript
EnsemblPlants_transcript
Cross_reference_EvolutionaryTrace
Cross_reference_ExpressionAtlas
Cross_reference_GeneID
Cross_reference_GeneTree
Cross_reference_Genevisible
Cross_reference_Gramene
Cross_reference_HOGENOM
Cross_reference_InParanoid
Cross_reference_IntAct
Cross_reference_InterPro
Cross_reference_iPTMnet
Cross_reference_jPOST
Cross_reference_KEGG
Cross_reference_MaxQB
Cross_reference_MetOSite
Cross_reference_MINT
Cross_reference_OMA
Cross_reference_OrthoDB
Cross_reference_PANTHER
Cross_reference_PATRIC
Cross_reference_PaxDb
Cross_reference_PDB
Cross_reference_PDBsum
Cross_reference_Pfam
Cross_reference_PhylomeDB
Cross_reference_PIR
Cross_reference_PRIDE
Cross_reference_PRO
Cross_reference_PROSITE
Cross_reference_ProteomicsDB
Cross_reference_RefSeq
Cross_reference_RNAct
Cross_reference_SABIO_RK
Cross_reference_SGD
Cross_reference_SMR
Cross_reference_STRING
Cross_reference_SUPFAM
Cross_reference_TAIR
Cross_reference_TIGRFAMs
Cross_reference_TubercuList
Cross_reference_UniPathway
Cross_reference_VEuPathDB
Non_AA_Character
AA_Character
UniProt_DB
MOTIF_SPVEY
MOTIF_RFDDR
truncated_seqs
truncated_seqs_latest
Length_2
Cross_reference_InterPro_2
</pre></div>
</div>
</div>
</div>
<div class="toctree-wrapper compound">
</div>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='right-next' id="next-link" href="als_interpro_IPR012782_subset_summary_cleaned.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">EC numbers</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By The Jupyter Book community<br/>
  
      &copy; Copyright 2022.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>