
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>EC numbers &#8212; kari_example_ec_1_1_1_86</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=5115cc725059bd94278eecd172e13a965bf8f5a9" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.b7bb847fb20b106c3d81b95245e65545.min.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=9c920249402e914e316237a7dbc6769907cce411"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="EC numbers" href="classI_kari_subset_summary_cleaned.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="index.html">
      
      
      
      <h1 class="site-logo" id="site-title">kari_example_ec_1_1_1_86</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1 current active">
                <a class="reference internal" href="#">
                    EC numbers
                </a>
            </li>
        </ul>
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="classI_kari_subset_summary_cleaned.html">
   EC numbers
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="eukaryotic_kari_subset_summary_cleaned.html">
   EC numbers
  </a>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
            <div class="navbar_extra_footer">
            Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
            </div>
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<div class="menu-dropdown menu-dropdown-launch-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Launch interactive content">
      <i class="fas fa-rocket"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="https://mybinder.org/v2/gh/executablebooks/jupyter-book/master?urlpath=tree/kari_example_ec_1_1_1_86_summary_cleaned.ipynb"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Launch on Binder"
>
  

<span class="headerbtn__icon-container">
  
    <img src="_static/images/logo_binder.svg">
  </span>
<span class="headerbtn__text-container">Binder</span>
</a>

      </li>
      
    </ul>
  </div>
</div>

<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>

<div class="menu-dropdown menu-dropdown-download-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Download this page">
      <i class="fas fa-download"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="_sources/kari_example_ec_1_1_1_86_summary_cleaned.ipynb"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Download source file"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="headerbtn__text-container">.ipynb</span>
</a>

      </li>
      
      <li>
        
<button onclick="printPdf(this)"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="left"
title="Print to PDF"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="headerbtn__text-container">.pdf</span>
</button>

      </li>
      
    </ul>
  </div>
</div>
<label for="__page-toc"
  class="headerbtn headerbtn-page-toc"
  
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-list"></i>
  </span>

</label>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
    <div class="tocsection onthispage pt-5 pb-3">
        <i class="fas fa-list"></i> Contents
    </div>
    <nav id="bd-toc-nav" aria-label="Page">
        <ul class="visible nav section-nav flex-column">
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#">
   EC numbers
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#taxonomic-distribution">
   Taxonomic distribution
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#annotation-distributions">
   Annotation distributions
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#experimental-data-brenda">
   Experimental data (BRENDA)
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#structural-information">
   Structural information
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#general-data-set-information">
   General data set information
  </a>
 </li>
</ul>

    </nav>
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>EC numbers</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#">
   EC numbers
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#taxonomic-distribution">
   Taxonomic distribution
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#annotation-distributions">
   Annotation distributions
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#experimental-data-brenda">
   Experimental data (BRENDA)
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#structural-information">
   Structural information
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#general-data-set-information">
   General data set information
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>None
</pre></div>
</div>
</div>
</div>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Summary sheet - kari_example_ec_1_1_1_86
</pre></div>
</div>
</div>
</div>
<section class="tex2jax_ignore mathjax_ignore" id="ec-numbers">
<h1>EC numbers<a class="headerlink" href="#ec-numbers" title="Permalink to this headline">#</a></h1>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>The EC numbers found in the current data set are {&#39;1.1.1.86&#39;}
</pre></div>
</div>
</div>
</div>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="taxonomic-distribution">
<h1>Taxonomic distribution<a class="headerlink" href="#taxonomic-distribution" title="Permalink to this headline">#</a></h1>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<img alt="_images/kari_example_ec_1_1_1_86_summary_cleaned_9_0.png" src="_images/kari_example_ec_1_1_1_86_summary_cleaned_9_0.png" />
<img alt="_images/kari_example_ec_1_1_1_86_summary_cleaned_9_1.png" src="_images/kari_example_ec_1_1_1_86_summary_cleaned_9_1.png" />
</div>
</div>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="annotation-distributions">
<h1>Annotation distributions<a class="headerlink" href="#annotation-distributions" title="Permalink to this headline">#</a></h1>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<img alt="_images/kari_example_ec_1_1_1_86_summary_cleaned_11_0.png" src="_images/kari_example_ec_1_1_1_86_summary_cleaned_11_0.png" />
<img alt="_images/kari_example_ec_1_1_1_86_summary_cleaned_11_1.png" src="_images/kari_example_ec_1_1_1_86_summary_cleaned_11_1.png" />
<img alt="_images/kari_example_ec_1_1_1_86_summary_cleaned_11_2.png" src="_images/kari_example_ec_1_1_1_86_summary_cleaned_11_2.png" />
<img alt="_images/kari_example_ec_1_1_1_86_summary_cleaned_11_3.png" src="_images/kari_example_ec_1_1_1_86_summary_cleaned_11_3.png" />
</div>
</div>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Sequence statistics for all sequences
Number of sequences : 24.0
Smallest sequnce length : 330.0
Longest sequnce length : 595.0
Average sequnce length : 396.6666666666667
</pre></div>
</div>
</div>
</div>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="experimental-data-brenda">
<h1>Experimental data (BRENDA)<a class="headerlink" href="#experimental-data-brenda" title="Permalink to this headline">#</a></h1>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>53
</pre></div>
</div>
<img alt="_images/kari_example_ec_1_1_1_86_summary_cleaned_14_1.png" src="_images/kari_example_ec_1_1_1_86_summary_cleaned_14_1.png" />
</div>
</div>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="structural-information">
<h1>Structural information<a class="headerlink" href="#structural-information" title="Permalink to this headline">#</a></h1>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>There are 13 total structures across 9 unique sequences
Looking at the first few...
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Entry is titled &quot;ketol-acid reductoisomerase (KARI) in complex with Mg2+&quot; and was released on 2009/04/14 

This entry was determined using X-ray diffraction 
The authors are Guddat, L.W.Leung, E.W.W. 

Assembly information -
Preferred form with assembly ID 1 is a homodimer
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Ligand : MAGNESIUM ION at position 1576
Ligand : MAGNESIUM ION at position 1577
Ligand : MAGNESIUM ION at position 1580
Ligand : MAGNESIUM ION at position 1597
Ligand : MAGNESIUM ION at position 1598
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>3fr7 chain A has helices at residue ranges [&#39;40-45&#39;, &#39;46-51&#39;, &#39;64-80&#39;, &#39;96-103&#39;, &#39;115-122&#39;, &#39;130-145&#39;, &#39;156-165&#39;, &#39;183-195&#39;, &#39;215-227&#39;, &#39;236-250&#39;, &#39;253-270&#39;, &#39;273-281&#39;, &#39;281-287&#39;, &#39;287-297&#39;, &#39;297-304&#39;, &#39;306-350&#39;, &#39;368-381&#39;, &#39;390-411&#39;, &#39;414-423&#39;, &#39;423-429&#39;, &#39;429-438&#39;, &#39;438-446&#39;, &#39;447-475&#39;, &#39;480-490&#39;, &#39;491-501&#39;] and strands at [&#39;24-29&#39;, &#39;32-38&#39;, &#39;56-60&#39;, &#39;85-90&#39;, &#39;112-114&#39;, &#39;124-127&#39;, &#39;150-153&#39;, &#39;174-180&#39;, &#39;204-210&#39;, &#39;231-234&#39;]
3fr7 chain B has helices at residue ranges [&#39;40-45&#39;, &#39;46-51&#39;, &#39;64-80&#39;, &#39;94-103&#39;, &#39;114-122&#39;, &#39;130-145&#39;, &#39;156-165&#39;, &#39;183-195&#39;, &#39;215-227&#39;, &#39;236-250&#39;, &#39;253-270&#39;, &#39;273-304&#39;, &#39;306-350&#39;, &#39;368-379&#39;, &#39;390-411&#39;, &#39;414-423&#39;, &#39;423-429&#39;, &#39;429-438&#39;, &#39;438-446&#39;, &#39;447-475&#39;, &#39;480-490&#39;, &#39;491-501&#39;] and strands at [&#39;25-29&#39;, &#39;32-38&#39;, &#39;56-60&#39;, &#39;85-89&#39;, &#39;112-113&#39;, &#39;124-127&#39;, &#39;150-153&#39;, &#39;174-180&#39;, &#39;204-210&#39;, &#39;231-234&#39;]
</pre></div>
</div>
<div class="output text_html"><div id="3dmolviewer_1664159806569188"  style="position: relative; width: 640px; height: 480px">
        <p id="3dmolwarning_1664159806569188" style="background-color:#ffcccc;color:black">You appear to be running in JupyterLab (or JavaScript failed to load for some other reason).  You need to install the 3dmol extension: <br>
        <tt>jupyter labextension install jupyterlab_3dmol</tt></p>
        </div>
<script>

var loadScriptAsync = function(uri){
  return new Promise((resolve, reject) => {
    var tag = document.createElement('script');
    tag.src = uri;
    tag.async = true;
    tag.onload = () => {
      resolve();
    };
  var firstScriptTag = document.getElementsByTagName('script')[0];
  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
});
};

if(typeof $3Dmolpromise === 'undefined') {
$3Dmolpromise = null;
  $3Dmolpromise = loadScriptAsync('https://cdn.jsdelivr.net/npm/3dmol@latest/build/3Dmol-min.min.js');
}

var viewer_1664159806569188 = null;
var warn = document.getElementById("3dmolwarning_1664159806569188");
if(warn) {
    warn.parentNode.removeChild(warn);
}
$3Dmolpromise.then(function() {
viewer_1664159806569188 = $3Dmol.createViewer($("#3dmolviewer_1664159806569188"),{backgroundColor:"white"});
$3Dmol.download("pdb:3FR7", viewer_1664159806569188, {}, function() {
viewer_1664159806569188.zoomTo();
	viewer_1664159806569188.setStyle({"cartoon": {"color": "spectrum"}});
viewer_1664159806569188.render();
})
});
</script></div><div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;py3Dmol.view at 0x119131fd0&gt;
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Entry is titled &quot;rice Ketolacid reductoisomerase in complex with Mg2+-NADPH&quot; and was released on 2009/04/14 

This entry was determined using X-ray diffraction 
The authors are Leung, E.W.W.Guddat, L.W. 

Assembly information -
Preferred form with assembly ID 1 is a homodimer
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Ligand : MAGNESIUM ION at position 1
Ligand : MAGNESIUM ION at position 1
Ligand : MAGNESIUM ION at position 597
Ligand : SULFATE ION at position 597
Ligand : NADPH DIHYDRO-NICOTINAMIDE-ADENINE-DINUCLEOTIDE PHOSPHATE at position 601
Ligand : NADPH DIHYDRO-NICOTINAMIDE-ADENINE-DINUCLEOTIDE PHOSPHATE at position 600
Ligand : 2-{2-[2-(2-{2-[2-(2-ETHOXY-ETHOXY)-ETHOXY]-ETHOXY}-ETHOXY)-ETHOXY]-ETHOXY}-ETHANOL at position 6061
Ligand : DI(HYDROXYETHYL)ETHER at position 6073
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>3fr8 chain A has helices at residue ranges [&#39;40-45&#39;, &#39;46-51&#39;, &#39;64-75&#39;, &#39;96-102&#39;, &#39;115-121&#39;, &#39;130-136&#39;, &#39;137-145&#39;, &#39;156-164&#39;, &#39;183-194&#39;, &#39;215-227&#39;, &#39;236-250&#39;, &#39;253-270&#39;, &#39;273-305&#39;, &#39;306-337&#39;, &#39;338-350&#39;, &#39;368-380&#39;, &#39;390-411&#39;, &#39;414-423&#39;, &#39;423-429&#39;, &#39;430-438&#39;, &#39;438-444&#39;, &#39;447-476&#39;, &#39;480-490&#39;, &#39;491-500&#39;] and strands at [&#39;25-29&#39;, &#39;32-38&#39;, &#39;56-60&#39;, &#39;85-90&#39;, &#39;112-114&#39;, &#39;124-127&#39;, &#39;150-153&#39;, &#39;174-180&#39;, &#39;204-210&#39;, &#39;231-234&#39;]
3fr8 chain B has helices at residue ranges [&#39;40-45&#39;, &#39;46-51&#39;, &#39;64-75&#39;, &#39;99-104&#39;, &#39;115-121&#39;, &#39;130-145&#39;, &#39;156-165&#39;, &#39;183-190&#39;, &#39;192-197&#39;, &#39;215-228&#39;, &#39;236-250&#39;, &#39;253-270&#39;, &#39;273-286&#39;, &#39;287-296&#39;, &#39;297-305&#39;, &#39;306-337&#39;, &#39;338-350&#39;, &#39;368-380&#39;, &#39;390-411&#39;, &#39;414-423&#39;, &#39;423-429&#39;, &#39;430-438&#39;, &#39;438-444&#39;, &#39;447-475&#39;, &#39;480-489&#39;, &#39;491-500&#39;] and strands at [&#39;25-29&#39;, &#39;32-38&#39;, &#39;56-60&#39;, &#39;85-90&#39;, &#39;112-114&#39;, &#39;124-127&#39;, &#39;150-153&#39;, &#39;174-180&#39;, &#39;204-210&#39;, &#39;231-234&#39;]
</pre></div>
</div>
<div class="output text_html"><div id="3dmolviewer_166415981007004"  style="position: relative; width: 640px; height: 480px">
        <p id="3dmolwarning_166415981007004" style="background-color:#ffcccc;color:black">You appear to be running in JupyterLab (or JavaScript failed to load for some other reason).  You need to install the 3dmol extension: <br>
        <tt>jupyter labextension install jupyterlab_3dmol</tt></p>
        </div>
<script>

var loadScriptAsync = function(uri){
  return new Promise((resolve, reject) => {
    var tag = document.createElement('script');
    tag.src = uri;
    tag.async = true;
    tag.onload = () => {
      resolve();
    };
  var firstScriptTag = document.getElementsByTagName('script')[0];
  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
});
};

if(typeof $3Dmolpromise === 'undefined') {
$3Dmolpromise = null;
  $3Dmolpromise = loadScriptAsync('https://cdn.jsdelivr.net/npm/3dmol@latest/build/3Dmol-min.min.js');
}

var viewer_166415981007004 = null;
var warn = document.getElementById("3dmolwarning_166415981007004");
if(warn) {
    warn.parentNode.removeChild(warn);
}
$3Dmolpromise.then(function() {
viewer_166415981007004 = $3Dmol.createViewer($("#3dmolviewer_166415981007004"),{backgroundColor:"white"});
$3Dmol.download("pdb:3FR8", viewer_166415981007004, {}, function() {
viewer_166415981007004.zoomTo();
	viewer_166415981007004.setStyle({"cartoon": {"color": "spectrum"}});
viewer_166415981007004.render();
})
});
</script></div><div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;py3Dmol.view at 0x1187a65e0&gt;
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Entry is titled &quot;Escherichia coli ketol-acid reductoisomerase&quot; and was released on 2005/10/11 

This entry was determined using X-ray diffraction 
The authors are Tyagi, R.Duquerroy, S.Navaza, J.Guddat, L.W.Duggleby, R.G. 

Assembly information -
Preferred form with assembly ID 1 is a homotetramer
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Ligand : SULFATE ION at position 4121
Ligand : SULFATE ION at position 4122
Ligand : SULFATE ION at position 4127
Ligand : SULFATE ION at position 4128
Ligand : SULFATE ION at position 4126
Ligand : SULFATE ION at position 4125
Ligand : SULFATE ION at position 4123
Ligand : SULFATE ION at position 4124
Ligand : SULFATE ION at position 4129
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>1yrl chain A has helices at residue ranges [&#39;4-8&#39;, &#39;9-18&#39;, &#39;24-29&#39;, &#39;32-36&#39;, &#39;46-59&#39;, &#39;68-74&#39;, &#39;76-85&#39;, &#39;92-96&#39;, &#39;97-99&#39;, &#39;107-122&#39;, &#39;133-138&#39;, &#39;159-167&#39;, &#39;179-182&#39;, &#39;186-199&#39;, &#39;199-204&#39;, &#39;209-223&#39;, &#39;226-243&#39;, &#39;246-276&#39;, &#39;278-309&#39;, &#39;310-321&#39;, &#39;324-335&#39;, &#39;337-342&#39;, &#39;350-357&#39;, &#39;358-378&#39;, &#39;382-390&#39;, &#39;393-413&#39;, &#39;414-438&#39;, &#39;454-468&#39;, &#39;468-487&#39;] and strands at [&#39;20-22&#39;, &#39;40-43&#39;, &#39;63-67&#39;, &#39;88-91&#39;, &#39;101-104&#39;, &#39;127-130&#39;, &#39;148-154&#39;, &#39;173-177&#39;, &#39;205-208&#39;]
1yrl chain B has helices at residue ranges [&#39;4-8&#39;, &#39;9-18&#39;, &#39;24-29&#39;, &#39;30-36&#39;, &#39;111-122&#39;, &#39;133-138&#39;, &#39;159-168&#39;, &#39;186-197&#39;, &#39;198-203&#39;, &#39;209-222&#39;, &#39;226-243&#39;, &#39;246-276&#39;, &#39;278-320&#39;, &#39;324-336&#39;, &#39;338-342&#39;, &#39;350-357&#39;, &#39;358-378&#39;, &#39;382-388&#39;, &#39;393-413&#39;, &#39;414-438&#39;, &#39;454-467&#39;, &#39;468-487&#39;] and strands at [&#39;20-22&#39;, &#39;101-103&#39;, &#39;127-130&#39;, &#39;148-154&#39;, &#39;173-177&#39;, &#39;205-208&#39;]
1yrl chain C has helices at residue ranges [&#39;4-8&#39;, &#39;9-18&#39;, &#39;24-29&#39;, &#39;46-61&#39;, &#39;68-74&#39;, &#39;76-85&#39;, &#39;92-96&#39;, &#39;107-122&#39;, &#39;133-139&#39;, &#39;159-167&#39;, &#39;179-182&#39;, &#39;186-198&#39;, &#39;199-203&#39;, &#39;209-223&#39;, &#39;226-243&#39;, &#39;246-276&#39;, &#39;278-322&#39;, &#39;324-335&#39;, &#39;337-342&#39;, &#39;350-357&#39;, &#39;358-378&#39;, &#39;382-390&#39;, &#39;393-413&#39;, &#39;414-432&#39;, &#39;432-437&#39;, &#39;454-467&#39;, &#39;468-487&#39;] and strands at [&#39;20-22&#39;, &#39;40-43&#39;, &#39;63-67&#39;, &#39;88-91&#39;, &#39;101-104&#39;, &#39;127-129&#39;, &#39;148-154&#39;, &#39;173-177&#39;, &#39;205-208&#39;]
1yrl chain D has helices at residue ranges [&#39;4-8&#39;, &#39;9-18&#39;, &#39;24-29&#39;, &#39;32-36&#39;, &#39;50-58&#39;, &#39;70-74&#39;, &#39;107-109&#39;, &#39;110-122&#39;, &#39;133-138&#39;, &#39;159-167&#39;, &#39;179-182&#39;, &#39;186-198&#39;, &#39;199-204&#39;, &#39;209-224&#39;, &#39;226-243&#39;, &#39;246-275&#39;, &#39;278-309&#39;, &#39;310-321&#39;, &#39;324-336&#39;, &#39;350-357&#39;, &#39;358-378&#39;, &#39;382-391&#39;, &#39;393-413&#39;, &#39;414-432&#39;, &#39;432-437&#39;, &#39;454-467&#39;, &#39;468-487&#39;] and strands at [&#39;21-22&#39;, &#39;39-42&#39;, &#39;62-65&#39;, &#39;101-103&#39;, &#39;127-130&#39;, &#39;148-154&#39;, &#39;173-177&#39;, &#39;205-207&#39;]
</pre></div>
</div>
<div class="output text_html"><div id="3dmolviewer_1664159813763365"  style="position: relative; width: 640px; height: 480px">
        <p id="3dmolwarning_1664159813763365" style="background-color:#ffcccc;color:black">You appear to be running in JupyterLab (or JavaScript failed to load for some other reason).  You need to install the 3dmol extension: <br>
        <tt>jupyter labextension install jupyterlab_3dmol</tt></p>
        </div>
<script>

var loadScriptAsync = function(uri){
  return new Promise((resolve, reject) => {
    var tag = document.createElement('script');
    tag.src = uri;
    tag.async = true;
    tag.onload = () => {
      resolve();
    };
  var firstScriptTag = document.getElementsByTagName('script')[0];
  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
});
};

if(typeof $3Dmolpromise === 'undefined') {
$3Dmolpromise = null;
  $3Dmolpromise = loadScriptAsync('https://cdn.jsdelivr.net/npm/3dmol@latest/build/3Dmol-min.min.js');
}

var viewer_1664159813763365 = null;
var warn = document.getElementById("3dmolwarning_1664159813763365");
if(warn) {
    warn.parentNode.removeChild(warn);
}
$3Dmolpromise.then(function() {
viewer_1664159813763365 = $3Dmol.createViewer($("#3dmolviewer_1664159813763365"),{backgroundColor:"white"});
$3Dmol.download("pdb:1YRL", viewer_1664159813763365, {}, function() {
viewer_1664159813763365.zoomTo();
	viewer_1664159813763365.setStyle({"cartoon": {"color": "spectrum"}});
viewer_1664159813763365.render();
})
});
</script></div><div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;py3Dmol.view at 0x119092190&gt;
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Entry is titled &quot;E. coli Ketol-acid reductoisomerase in complex with NADPH and Mg2+&quot; and was released on 2012/10/17 

This entry was determined using X-ray diffraction 
The authors are Wong, S.H.Lonhienne, T.G.A.Winzor, D.J.Schenk, G.Guddat, L.W. 

Assembly information -
Preferred form with assembly ID 1 is a homotetramer
Non-preferred form with assembly ID 2 is a homotetramer
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Ligand : NADPH DIHYDRO-NICOTINAMIDE-ADENINE-DINUCLEOTIDE PHOSPHATE at position 492
Ligand : NADPH DIHYDRO-NICOTINAMIDE-ADENINE-DINUCLEOTIDE PHOSPHATE at position 492
Ligand : SULFATE ION at position 493
Ligand : SULFATE ION at position 494
Ligand : SULFATE ION at position 495
Ligand : SULFATE ION at position 496
Ligand : SULFATE ION at position 497
Ligand : SULFATE ION at position 493
Ligand : SULFATE ION at position 494
Ligand : SULFATE ION at position 495
Ligand : SULFATE ION at position 496
Ligand : MAGNESIUM ION at position 498
Ligand : MAGNESIUM ION at position 499
Ligand : MAGNESIUM ION at position 497
Ligand : MAGNESIUM ION at position 498
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>3ulk chain A has helices at residue ranges [&#39;4-8&#39;, &#39;9-18&#39;, &#39;24-29&#39;, &#39;32-36&#39;, &#39;46-59&#39;, &#39;68-74&#39;, &#39;76-85&#39;, &#39;92-96&#39;, &#39;97-99&#39;, &#39;107-109&#39;, &#39;110-119&#39;, &#39;120-122&#39;, &#39;133-138&#39;, &#39;157-167&#39;, &#39;179-182&#39;, &#39;186-198&#39;, &#39;199-203&#39;, &#39;209-223&#39;, &#39;226-243&#39;, &#39;246-276&#39;, &#39;278-322&#39;, &#39;324-337&#39;, &#39;337-342&#39;, &#39;350-357&#39;, &#39;358-378&#39;, &#39;382-389&#39;, &#39;390-392&#39;, &#39;393-413&#39;, &#39;414-432&#39;, &#39;432-438&#39;, &#39;454-467&#39;, &#39;468-488&#39;] and strands at [&#39;21-22&#39;, &#39;39-43&#39;, &#39;62-67&#39;, &#39;88-91&#39;, &#39;101-104&#39;, &#39;127-130&#39;, &#39;148-154&#39;, &#39;173-177&#39;, &#39;205-207&#39;]
3ulk chain B has helices at residue ranges [&#39;4-8&#39;, &#39;9-18&#39;, &#39;24-29&#39;, &#39;32-36&#39;, &#39;46-59&#39;, &#39;68-74&#39;, &#39;76-85&#39;, &#39;92-96&#39;, &#39;97-99&#39;, &#39;107-109&#39;, &#39;110-119&#39;, &#39;120-122&#39;, &#39;133-138&#39;, &#39;157-167&#39;, &#39;179-182&#39;, &#39;186-199&#39;, &#39;200-203&#39;, &#39;209-223&#39;, &#39;226-243&#39;, &#39;246-276&#39;, &#39;278-322&#39;, &#39;324-336&#39;, &#39;337-342&#39;, &#39;350-357&#39;, &#39;358-378&#39;, &#39;382-389&#39;, &#39;390-392&#39;, &#39;393-413&#39;, &#39;414-432&#39;, &#39;432-438&#39;, &#39;454-467&#39;, &#39;468-489&#39;] and strands at [&#39;21-22&#39;, &#39;39-43&#39;, &#39;62-67&#39;, &#39;88-91&#39;, &#39;101-104&#39;, &#39;127-130&#39;, &#39;148-154&#39;, &#39;173-177&#39;, &#39;205-207&#39;]
</pre></div>
</div>
<div class="output text_html"><div id="3dmolviewer_166415981754387"  style="position: relative; width: 640px; height: 480px">
        <p id="3dmolwarning_166415981754387" style="background-color:#ffcccc;color:black">You appear to be running in JupyterLab (or JavaScript failed to load for some other reason).  You need to install the 3dmol extension: <br>
        <tt>jupyter labextension install jupyterlab_3dmol</tt></p>
        </div>
<script>

var loadScriptAsync = function(uri){
  return new Promise((resolve, reject) => {
    var tag = document.createElement('script');
    tag.src = uri;
    tag.async = true;
    tag.onload = () => {
      resolve();
    };
  var firstScriptTag = document.getElementsByTagName('script')[0];
  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
});
};

if(typeof $3Dmolpromise === 'undefined') {
$3Dmolpromise = null;
  $3Dmolpromise = loadScriptAsync('https://cdn.jsdelivr.net/npm/3dmol@latest/build/3Dmol-min.min.js');
}

var viewer_166415981754387 = null;
var warn = document.getElementById("3dmolwarning_166415981754387");
if(warn) {
    warn.parentNode.removeChild(warn);
}
$3Dmolpromise.then(function() {
viewer_166415981754387 = $3Dmol.createViewer($("#3dmolviewer_166415981754387"),{backgroundColor:"white"});
$3Dmol.download("pdb:3ULK", viewer_166415981754387, {}, function() {
viewer_166415981754387.zoomTo();
	viewer_166415981754387.setStyle({"cartoon": {"color": "spectrum"}});
viewer_166415981754387.render();
})
});
</script></div><div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;py3Dmol.view at 0x119131fd0&gt;
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Entry is titled &quot;Crystal structure of Mycobacterium tuberculosis ketol-acid reductoisomerase in complex with Mg2+&quot; and was released on 2016/02/17 

This entry was determined using X-ray diffraction 
The authors are Lv, Y.Guddat, L.W. 

Assembly information -
Preferred form with assembly ID 1 is a homodimer
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Ligand : MAGNESIUM ION at position 401
Ligand : MAGNESIUM ION at position 402
Ligand : MAGNESIUM ION at position 403
Ligand : MAGNESIUM ION at position 404
Ligand : MAGNESIUM ION at position 406
Ligand : MAGNESIUM ION at position 407
Ligand : MAGNESIUM ION at position 408
Ligand : MAGNESIUM ION at position 410
Ligand : MAGNESIUM ION at position 401
Ligand : MAGNESIUM ION at position 402
Ligand : MAGNESIUM ION at position 403
Ligand : MAGNESIUM ION at position 404
Ligand : MAGNESIUM ION at position 405
Ligand : CHLORIDE ION at position 405
Ligand : CHLORIDE ION at position 409
Ligand : SODIUM ION at position 406
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>4ypo chain A has helices at residue ranges [&#39;3-7&#39;, &#39;9-14&#39;, &#39;23-36&#39;, &#39;50-57&#39;, &#39;63-71&#39;, &#39;79-81&#39;, &#39;82-90&#39;, &#39;91-95&#39;, &#39;106-111&#39;, &#39;130-140&#39;, &#39;157-169&#39;, &#39;170-175&#39;, &#39;180-195&#39;, &#39;198-214&#39;, &#39;217-226&#39;, &#39;228-248&#39;, &#39;249-264&#39;, &#39;265-281&#39;, &#39;282-294&#39;, &#39;296-308&#39;, &#39;309-320&#39;] and strands at [&#39;16-20&#39;, &#39;39-43&#39;, &#39;60-62&#39;, &#39;73-76&#39;, &#39;100-103&#39;, &#39;121-127&#39;, &#39;146-152&#39;, &#39;176-178&#39;]
4ypo chain B has helices at residue ranges [&#39;3-7&#39;, &#39;9-14&#39;, &#39;22-24&#39;, &#39;25-36&#39;, &#39;50-57&#39;, &#39;63-71&#39;, &#39;79-81&#39;, &#39;82-91&#39;, &#39;92-95&#39;, &#39;106-111&#39;, &#39;130-140&#39;, &#39;157-169&#39;, &#39;170-175&#39;, &#39;180-195&#39;, &#39;198-214&#39;, &#39;217-226&#39;, &#39;228-248&#39;, &#39;249-264&#39;, &#39;265-281&#39;, &#39;282-294&#39;, &#39;296-308&#39;, &#39;309-320&#39;] and strands at [&#39;16-20&#39;, &#39;39-43&#39;, &#39;60-62&#39;, &#39;73-76&#39;, &#39;100-103&#39;, &#39;121-127&#39;, &#39;146-152&#39;, &#39;176-178&#39;]
</pre></div>
</div>
<div class="output text_html"><div id="3dmolviewer_1664159821252822"  style="position: relative; width: 640px; height: 480px">
        <p id="3dmolwarning_1664159821252822" style="background-color:#ffcccc;color:black">You appear to be running in JupyterLab (or JavaScript failed to load for some other reason).  You need to install the 3dmol extension: <br>
        <tt>jupyter labextension install jupyterlab_3dmol</tt></p>
        </div>
<script>

var loadScriptAsync = function(uri){
  return new Promise((resolve, reject) => {
    var tag = document.createElement('script');
    tag.src = uri;
    tag.async = true;
    tag.onload = () => {
      resolve();
    };
  var firstScriptTag = document.getElementsByTagName('script')[0];
  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
});
};

if(typeof $3Dmolpromise === 'undefined') {
$3Dmolpromise = null;
  $3Dmolpromise = loadScriptAsync('https://cdn.jsdelivr.net/npm/3dmol@latest/build/3Dmol-min.min.js');
}

var viewer_1664159821252822 = null;
var warn = document.getElementById("3dmolwarning_1664159821252822");
if(warn) {
    warn.parentNode.removeChild(warn);
}
$3Dmolpromise.then(function() {
viewer_1664159821252822 = $3Dmol.createViewer($("#3dmolviewer_1664159821252822"),{backgroundColor:"white"});
$3Dmol.download("pdb:4YPO", viewer_1664159821252822, {}, function() {
viewer_1664159821252822.zoomTo();
	viewer_1664159821252822.setStyle({"cartoon": {"color": "spectrum"}});
viewer_1664159821252822.render();
})
});
</script></div><div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;py3Dmol.view at 0x119092460&gt;
</pre></div>
</div>
</div>
</div>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="general-data-set-information">
<h1>General data set information<a class="headerlink" href="#general-data-set-information" title="Permalink to this headline">#</a></h1>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<p>These are the columns that are available and have at least one entry</p>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>accession
id
gene_names
gene_primary
gene_synonym
gene_oln
gene_orf
organism_name
organism_id
protein_name
sequence
xref_proteomes
lineage
cc_alternative_products
cc_sequence_caution
length
mass
ft_conflict
sequence_version
ec
cc_catalytic_activity
cc_cofactor
cc_function
kinetics
cc_pathway
ph_dependence
ft_act_site
ft_binding
annotation_score
feature_count
cc_caution
cc_miscellaneous
keyword
protein_existence
reviewed
cc_subunit
cc_induction
go
go_p
go_f
go_c
go_id
cc_subcellular_location
ft_chain
ft_init_met
ft_mod_res
ft_transit
structure_3d
ft_strand
ft_helix
ft_turn
lit_pubmed_id
date_created
date_modified
date_sequence_modified
version
protein_families
ft_compbias
ft_domain
ft_region
lineage.1
xref_araport
xref_bindingdb
xref_biocyc
xref_biogrid
xref_brenda
xref_carbonyldb
xref_chembl
xref_dip
xref_dnasu
xref_echobase
xref_eggnog
xref_embl
xref_ensemblbacteria
xref_ensemblfungi
xref_ensemblplants
xref_evolutionarytrace
xref_expressionatlas
xref_gene3d
xref_geneid
xref_genevisible
xref_gramene
xref_hamap
xref_hogenom
xref_inparanoid
xref_intact
xref_interpro
xref_iptmnet
xref_jpost
xref_kegg
xref_leproma
xref_maxqb
xref_mint
xref_moonprot
xref_oma
xref_orthodb
xref_panther
xref_patric
xref_paxdb
xref_pdb
xref_pdbsum
xref_pfam
xref_phylomedb
xref_pir
xref_pirsf
xref_plantreactome
xref_pombase
xref_pride
xref_pro
xref_prosite
xref_proteomicsdb
xref_pseudocap
xref_refseq
xref_rnact
xref_sabio_rk
xref_sgd
xref_smr
xref_string
xref_supfam
xref_swiss_2dpage
xref_tair
xref_tigrfams
xref_topdownproteomics
xref_tuberculist
xref_unipathway
xref_veupathdb
xref_world_2dpage
comment_count
lineage_superkingdom
lineage_kingdom
lineage_subkingdom
lineage_phylum
lineage_subphylum
lineage_class
lineage_subclass
lineage_order
lineage_family
lineage_subfamily
lineage_tribe
lineage_subtribe
lineage_genus
Non_AA_Character
AA_Character
Length_2
Cross_reference_InterPro_2
KARI_Class
psuedomonas_aeruginosa
NADH_pref
General_approach_Class_I
</pre></div>
</div>
</div>
</div>
<div class="toctree-wrapper compound">
</div>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='right-next' id="next-link" href="classI_kari_subset_summary_cleaned.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">EC numbers</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By The Jupyter Book community<br/>
  
      &copy; Copyright 2022.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>